src/Cones/hypoperlogdettri.jl

```diff
@@ -319,89 +331,77 @@ function inv_hess_prod!(
+        @views prod_w = prod[3:end, j]
+        rw_const = dot(R, W)
+        qγr = q + rw_const / (ζ + v)
+        cv = c4 * (c0 * p + qγr)
+        prod[1, j] = cone.Hiuu * p + c4 * c0 * qγr + rw_const * α_const
+        prod[2, j] = cv
+        w_prod = (p * α_const + cv / (ζ + v)) * W + W * R * W / (v * ζi + 1)
+        @views smat_to_svec!(prod[3:end, j], w_prod, cone.rt2)
     end
-    @inbounds @views mul!(prod[3:end, :], Hi[1:2, 3:end]', arr[1:2, :],
-        true, z / zv)
 
     return prod
 end
 
 function dder3(cone::HypoPerLogdetTri, dir::AbstractVector)
     @assert cone.grad_updated
-    u_dir = dir[1]
-    v_dir = dir[2]
-    @views w_dir = dir[3:end]
-    dder3 = cone.dder3
-    @views w_dder3 = dder3[3:end]
+    p = dir[1]
+    q = dir[2]
+    @views r = dir[3:end]
+    corr = cone.correction
     v = cone.point[2]
-    z = cone.z
-    tempw = cone.tempw
-    Wi_vec = cone.Wi_vec
+    ζ = cone.ζ
+    ζi = cone.ζi
+    σ = cone.σ
     d = cone.d
-    dlzi = (d - cone.lwv) / z
-    vz = v / z
-    udz = u_dir / z
-    vdz = v_dir / z
-
-    dot_Wi_S = dot(Wi_vec, w_dir)
-    S = copytri!(svec_to_smat!(cone.mat2, w_dir, cone.rt2), 'U', true)
-    ldiv!(cone.fact_W, S)
-    dot_skron = real(dot(S, S'))
-
-    const6 = 1 + v * dlzi
-    uuw_scal = -2 * udz * vz / z
-    vvw_scal = -vdz * (2 * const6 * dlzi + d / z)
-    uvw_scal = -2 * (2 * vz * dlzi + inv(z)) / z
-    const8 = -2 * (1 + vz)
-    const10 = 2 * (vz * udz + vdz * const6)
-    const9 = -2 * abs2(vz) * dot_Wi_S + const10
-    const7 = uuw_scal + uvw_scal * v_dir
-    const11 = -vz * (vz * dot_skron + 2 * (abs2(vz * dot_Wi_S) - dot_Wi_S *
-        const10)) + u_dir * const7 + vvw_scal * v_dir
-
-    @. w_dder3 = const11 * Wi_vec
-    rdiv!(S, cone.fact_W.U)
-    mul!(cone.mat3, S, S')
-    vec_S2 = smat_to_svec!(tempw, cone.mat3, cone.rt2)
-    @. w_dder3 += const8 * vec_S2
-    skron2 = rdiv!(S, cone.fact_W.U')
-    vec_skron2 = smat_to_svec!(tempw, skron2, cone.rt2)
-    t4awd = (2 * vz * abs2(dot_Wi_S) + dot(vec_skron2, w_dir)) / z
-    @. w_dder3 += const9 * vec_skron2
-
-    const2 = 2 * udz * dlzi / z
-    const3 = (2 * abs2(dlzi) + d / z / v) * vdz
-    const5 = d / v / z
-    const4 = dlzi * (2 * abs2(dlzi) + 3 * const5) - (const5 + 2 * inv(v) / v) / v
-    dder3[1] = 2 * abs2(udz) / z + (2 * const2 + const3) * v_dir +
-        (uuw_scal + const7) * dot_Wi_S + vz * t4awd
-    dder3[2] = const4 * abs2(v_dir) + (2 * vvw_scal + uvw_scal * u_dir) *
-        dot_Wi_S + u_dir * (const2 + 2 * const3) + const6 * t4awd
-
-    dder3 ./= -2
+    Wi = Hermitian(cone.Wi)
+    @views w = cone.point[3:end]
+    W = Hermitian(svec_to_smat!(cone.W, w, cone.rt2), :U)
+    viq = q / v
+    viq2 = abs2(viq)
+    w_aux = cone.mat5
+
+    r_X = copytri!(svec_to_smat!(cone.mat3, r, cone.rt2), 'U', cone.is_complex)
+    c0 = dot(Wi, r_X)
+    ∇ϕr = c0 * v
+    χ = -p + σ * q + ∇ϕr
+
+    rwi = rdiv!(r_X, cone.fact_W)
+    rwi_sqr = dot(rwi, rwi')
+    L_rwi = ldiv!(cone.fact_W.L, rwi)
```

Comment:
I think it is better to use `cone.fact_W.U'` here, because getting the L is less efficient than getting the U since the cholesky is a U cholesky.